inference-engine/src/readers/onnx_reader/ie_onnx_reader.cpp

```diff
@@ -65,12 +65,6 @@ CNNNetwork ONNXReader::read(std::istream& model, const std::vector<IExtensionPtr
     return CNNNetwork(ngraph::onnx_import::import_onnx_model(model, readPathFromStream(model)), exts);
 }
 
-INFERENCE_PLUGIN_API(StatusCode) InferenceEngine::CreateReader(IReader*& reader, ResponseDesc *resp) noexcept {
-    try {
-        reader = new ONNXReader();
-        return OK;
-    }
-    catch (std::exception &) {
-        return GENERAL_ERROR;
-    }
+INFERENCE_PLUGIN_API(void) InferenceEngine::CreateReader(std::shared_ptr<IReader>& reader) {
+    reader = std::make_shared<ONNXReader>();
```

Comment:
can we return as return value?

Reply:
No. We should use C API here (without mangling). It Dose not support such type as `std::shared_ptr` as return type.

Reply:
I have compiled:
```
INFERENCE_EXTENSION_API(std::shared_ptr<InferenceEngine::IExtension>) InferenceEngine::CreateExtensionShared() {
    return std::make_shared<Extension>();
}
```
And I have:
```
$ readelf -Ws lib/libtemplate_extension.so | grep CreateExtension
   167: 0000000000010220   211 FUNC    GLOBAL DEFAULT   11 CreateExtensionShared
   198: 00000000000101b0   112 FUNC    GLOBAL DEFAULT   11 CreateExtension
```

PS: Not sure about Windows..